File: two-sum/two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    map = Hash.new
    
    nums.each_with_index { |val, index|
        if map.key? (target - val)
            return [map[target-val], index]
        end
        map[val] = index
    }
    
end

File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap();
        
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                
                int posn = map.get(target - nums[i]);
                return new int[] {i, posn};
            }
            
            map.put(nums[i], i);   
        }
        
        return new int[] {};
    }
}